app/queries/bestsellers_query.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class BestsellersQuery
+  def self.call
+    sql = %[SELECT DISTINCT ON (bc.category_id) b.*, SUM(oi.quantity) as total_quantity
+      FROM order_items oi JOIN
+          orders o
+          ON oi.order_id = o.id JOIN
+          books b
+          ON oi.book_id = b.id
+      INNER JOIN books_categories bc ON bc.book_id = b.id
+      WHERE o.status in (2, 3)
+      GROUP BY bc.category_id, b.id
+      ORDER BY bc.category_id, total_quantity DESC]
+
+    Book.find_by_sql(sql)
+  end
+end
```

Comment:
- first: try to not use raw sql queries. Orm can solve many of your tasks
- second: you can create one query class for all cases with bookings queries. Put some **filter** attribute into this class to choose the case (popular, all, etc)
Useful docs fot implementation: 
[yield_self](https://apidock.com/ruby/v2_5_5/Object/yield_self) 
[&method](https://til.hashrocket.com/posts/2dab9b4db4-ruby-array-shortcuts-and-method)

Reply:
I see you implement this in app/queries/default_books_query.rb
Just rename to BooksQuery and all new filter method

Reply:
read about delegate